src/components/FileUpload.tsx

```diff
@@ -54,16 +55,105 @@ export default function FileUpload() {
+                         addDataSet(dataSetKey:$datasetKey){
+                            id
+                            name
+                            datasets
+                        }
+                }
+                `,
+                variables: {
+                    datasetKey: `${process.env.NEXT_PUBLIC_DATASET_API}/${key}`
+                }
+            })
+        })
+            .then((resp) => resp.json())
+            .then((result) => {
+                enqueueSnackbar("Dataset uploaded", {
+                    variant: "success"
+                });
+            })
+            .catch((err) => console.log(err));
+    }
+
     return (
         <>
             <Dropzone
                 onDrop={async (acceptedFiles: File[]) => {
-                    console.log("start");
                     let dsChanged = false;
+                    let fileToUploaded = new FormData();
+                    fileToUploaded.append("upload", acceptedFiles[0]);
+
+                    fetch(process.env.NEXT_PUBLIC_DATASET_API, {
+                        method: "POST",
+                        headers: {
+                            //TODO (verite) I'm sure there is a better way to write this but I'll fix this letter
+                            Authorization:
+                                (await getToken()) as unknown as string
+                        },
+                        body: fileToUploaded
+                    })
+                        .then((response) => response.json())
+                        .then((resp) => {
+                            if (resp.message === "Invalid auth token") {
+                                enqueueSnackbar(
+                                    "Session expired, login to upload the dataset",
+                                    {
+                                        variant: "error",
+                                        preventDuplicate: true
+                                    }
+                                );
+                            }
+
+                            if (resp.message === "Dataset already exists") {
+                                enqueueSnackbar(resp.message, {
+                                    variant: "error"
+                                });
+                                return;
+                            }
+
+                            console.log(resp);
+
+                            uploadDatatset(
+                                resp.Key + " &#$@ " + acceptedFiles[0].name
+                            );
+                        })
+                        .catch((error) => {
+                            // Handle sending the file key to the graphql api
+                            enqueueSnackbar("Dataset upload failed", {
+                                variant: "error"
+                            });
+                            return;
+                        });
+
                     await Promise.all(
```

Comment:
We need to integrate this code with the dataset uploading part. Previously the way this worked is uploading a file would load it into an array (datasets) and then call `setDataBlockEnabled` which enables the block in the workspace (I think we can remove this bit if the uploads will always be available to all users). We need to figure out a different way to make datasets available to all parts of the code.

Another thing to consider, what happens if a project is public and someone who isn't the user who made the project tries to use the dataset? It should let them.